bft/bft_interp/src/lib.rs

```diff
@@ -305,13 +321,82 @@ where
     /// interp.write_value(&mut data);
     /// assert_eq!(data.get_ref()[1], 7);
     /// ```
-    pub fn write_value(&mut self, file: &mut impl Write) -> Result<(), VMError> {
+    pub fn write_value(&mut self, file: &mut impl Write) -> Result<usize, VMError> {
         let mut buffer: [u8; 1] = [0; 1];
-        buffer[0] = self.cells[self.head].get_value();
-        file.write_all(&buffer).map_err(|e| VMError::IOError {
-            instruction: self.current_instruction().instruction(),
-            source: e,
-        })
+        buffer[0] = self.current_cell().get_value();
+        file.write_all(&buffer)
+            .and_then(|_| file.flush())
+            .map_err(|e| VMError::IOError {
+                instruction: self.current_instruction().instruction(),
+                source: e,
+            })
+            .and(Ok(self.next_instruction()))
+    }
+
+    pub fn open_loop(&mut self) -> Result<usize, VMError> {
+        if self.current_cell().is_zero() {
+            assert!(matches!(
+                self.current_instruction(),
+                DecoratedInstruction::OpenLoop { .. }
+            ));
```

Comment:
Surely this function won't be called unless the current instruction is an openloop?